.github/workflows/pull-request.yaml

```diff
@@ -1,38 +1,46 @@
 # This workflow will lint the PR's title and commit message.
-# https://github.com/CondeNast/conventional-pull-request-action
-# https://github.com/conventional-changelog/commitlint/blob/master/docs/reference-rules.md
-# https://github.com/CondeNast/conventional-pull-request-action/issues/15
 
 name: Pull Request
 on:
   pull_request:
-    branches: [main, staging]
+    branches:
+    - main
+    - staging
     types:
     - opened
+    - reopened
     - edited
     - synchronize
 permissions:
   contents: read
 
 jobs:
-  pr:
-    permissions:
-      pull-requests: read  # for CondeNast/conventional-pull-request-action to get specific PR & list commits
-    name: Lint PR title
+  conventional-commits:
+    name: Check PR title and commit messages
     runs-on: ubuntu-latest
     steps:
-    - name: Harden Runner
-      uses: step-security/harden-runner@74b568e8591fbb3115c70f3436a0c6b0909a8504
+    - name: Check out repository
+      uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b # v3.0.2
       with:
-        egress-policy: audit # TODO: change to 'egress-policy: block' after couple of runs
-
-    - name: Run conventional commits check
-      uses: CondeNast/conventional-pull-request-action@eef02ae6e88e4b462f39931113b1a75cc275beee
-      env:
-        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        fetch-depth: 0
+    - name: Set up Python
+      uses: actions/setup-python@c4e89fac7e8767b327bbad6cb4d859eda999cf08 # v4.1.0
       with:
-        commitlintRulesPath: ./commitlint.rules.js
-        commitTitleMatch: 'true'
+        python-version: '3.10'
+    - name: Set up Commitizen
+      run: |
+        pip install --upgrade pip wheel
+        pip install 'commitizen ==2.28.0'
```

Comment:
Can you add comments to explain how these steps work?

Reply:
Like this: commit https://github.com/jenstroeger/python-package-template/pull/263/commits/aa5b0eebd273af52da61ac9c5e08da986cd0a1ee?

Reply:
I meant to explain why you do `pip install 'commitizen ==2.28.0'` directly for example instead of using `make setup`.

Reply:
Commit https://github.com/jenstroeger/python-package-template/pull/263/commits/0a1d3cdf48b3e7b18360d6d3264a4ef6759d54c8.